Derpy/Utils/Tumblr/Model.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+
+namespace Derpy.Utils.Tumblr
+{
+    public class Post
```

Comment:
Very uninformed take here, but wouldn't all these be better as simple `struct`s and properties be simple members? Is there a reason to do this way?

Reply:
`struct`s are a nice idea - I was a bit concerned that having `string` fields would mess with their sort of compact, atomic nature, but size-wise a `string` actually just behave as a pointer to an internal string table. The properties have to be as they are or else the deserialization code cannot properly set the field values, and so you end up with a whole lot of null reference exceptions. I'm mainly going off [this web page](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct) in my decision on this though, haha. Bear in mind that page recommends to use `class`es unless some criteria are met, but aside from boxing (which I'm completely lost on), I think we could use them here? Yeah, C# `struct`s behave slightly differently from other languages I've used them in due to the garbage collection, so I'm a bit unsure on the nitty-gritty.